app/models/game.rb

```diff
@@ -87,6 +87,108 @@ def is_in_check?(player)
+      black_player.pieces.where(game: self, is_active: true).each do |piece|
+        return false if valid_check_block?(threatening_piece, piece, king) && piece.type != "King"
+      end
+
+      return true
+    end
+  end
+
+  def valid_check_block?(threatening_piece, piece, king)
+    threatening_piece_x = threatening_piece.position_x
+    threatening_piece_y = threatening_piece.position_y
+    king_x = king.position_x
+    king_y = king.position_y
+
+    if threatening_piece.is_diagonal_move?(king_x, king_y)
+      if threatening_piece_x < king_x && threatening_piece_y < king_y # Northeast
+        (threatening_piece_x + 1).upto(king_x - 1) do |x|
+          (threatening_piece_y + 1).upto(king_y - 1) do |y|
+            return true if piece.is_valid_move?(x, y) && (x - threatening_piece.position_x).abs == (y - threatening_piece.position_y).abs
+          end 
+        end 
+      elsif threatening_piece_x > king_x && threatening_piece_y < king_y # Northwest
+        (threatening_piece_x - 1).downto(king_x + 1) do |x|
+          (threatening_piece_y + 1).upto(king_y - 1) do |y|
+            return true if piece.is_valid_move?(x, y) && (x - threatening_piece.position_x).abs == (y - threatening_piece.position_y).abs
+          end 
+        end 
+      elsif threatening_piece_x < king_x && threatening_piece_y > king_y # Southeast
+        (threatening_piece_x + 1).upto(king_x - 1) do |x|
+          (threatening_piece_y - 1).downto(king_y + 1) do |y|
+            return true if piece.is_valid_move?(x, y) && (x - threatening_piece.position_x).abs == (y - threatening_piece.position_y).abs
+          end 
+        end 
+      elsif threatening_piece_x > king_x && threatening_piece_y > king_y # Southwest
+        (threatening_piece_x - 1).downto(king_x + 1) do |x|
+          (threatening_piece_y - 1).downto(king_y + 1) do |y|
+            return true if piece.is_valid_move?(x, y) && (x - threatening_piece.position_x).abs == (y - threatening_piece.position_y).abs
+          end 
+        end
+      end  
+    elsif threatening_piece.is_horizontal_move?(king_x, king_y)
+      if threatening_piece_x < king_x # East
+        (threatening_piece_x + 1).upto(king_x - 1) do |x|
+          return true if piece.is_valid_move?(x, king_y)
+        end 
+      else # West
+        (threatening_piece_x - 1).downto(king_x + 1) do |x|
+          return true if piece.is_valid_move?(x, king_y)
+        end 
+      end 
+    elsif threatening_piece.is_vertical_move?(king_x, king_y)
+      if threatening_piece_y < king_y # North
+        (threatening_piece_y + 1).upto(king_y - 1) do |y|
+          return true if piece.is_valid_move?(king_x, y)
+        end 
+      else # South
+        (threatening_piece_y - 1).downto(king_y + 1) do |y|
+          return true if piece.is_valid_move?(king_x, y)
+        end 
+      end 
+    elsif threatening_piece.is_knight_move?(king_x, king_y)
+      return false
+    end
+    false
+  end 
```

Comment:
If you find your methods getting this long, you might want to break them up into smaller methods, or find ways to use different logic to shrink them down. This method does not look DRY, just based on its structure.